### PR TITLE
fix(application): update configure_services

### PR DESCRIPTION
**Pull request by lengau**
- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `tox run -m lint`?
- [ ] Have you successfully run `tox run -e test-py310`? (supported versions: `py39`, `py310`, `py311`, `py312`)

-----

### CODE REVIEW

File: requirements-devel.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ click==8.1.7
 codespell==2.2.6
 colorama==0.4.6
 coverage==7.4.0
-craft-application @ git+https://github.com/canonical/craft-application@main
+craft-application @ git+https://github.com/canonical/craft-application@configure-services-better
```
Comment (lengau) [inline requirements-devel.txt:14]: Undo after merging https://github.com/canonical/craft-application/pull/259

File: requirements.txt [hunk 2]
```diff
@@ -5,7 +5,7 @@ cffi==1.16.0
 chardet==5.2.0
 charset-normalizer==3.3.2
 click==8.1.7
-craft-application @ git+https://github.com/canonical/craft-application@main
+craft-application @ git+https://github.com/canonical/craft-application@configure-services-better
```
Comment (lengau) [inline requirements.txt:8]: Undo after merging https://github.com/canonical/craft-application/pull/259

**Comment by mr-cal** [comment]
https://github.com/canonical/craft-application/pull/259 and https://github.com/canonical/craft-application/pull/242 both made breaking changes.  Some changes are fixed here and some are fixed in #4616.  Perhaps we should combine the PRs and do a merge commit?

**Comment by mr-cal** [comment]
Closing because the commits here were merged in via #4659.

**Pull request closed without merging**